internal/venice-common/src/main/java/com/linkedin/venice/writer/VeniceWriter.java

```diff
@@ -981,55 +981,58 @@ public CompletableFuture<PubSubProduceResult> put(
-    int totalRecordSize = calculateTotalRecordSize(serializedKey, serializedValue, null);
 
-    if (isChunkingNeededForRecord(totalRecordSize)) {
-      return sendDivMessageChunked(
-          partition,
-          serializedKey,
-          serializedValue,
-          GLOBAL_DIV_STATE.getCurrentProtocolVersion(),
-          null);
-    } else {
-      return sendDivMessageNonChunked(
-          partition,
-          serializedKey,
-          serializedValue,
-          GLOBAL_DIV_STATE.getCurrentProtocolVersion(),
-          null);
+    if (isChunkingNeededForRecord(serializedKey.length + serializedValue.length)) {
+      return sendChunkedGlobalRtDivMessage(partition, serializedKey, serializedValue);
     }
+
+    serializedKey = keyWithChunkingSuffixSerializer.serializeNonChunkedKey(serializedKey);
+    KafkaKey divKey = new KafkaKey(MessageType.GLOBAL_RT_DIV, serializedKey);
+
+    // Initialize the SpecificRecord instances used by the Avro-based Kafka protocol
+    Put putPayload =
+        buildPutPayload(serializedValue, AvroProtocolDefinition.GLOBAL_RT_DIV_STATE.getCurrentProtocolVersion(), null);
+
+    // TODO: This needs to be implemented later to support Global RT DIV
+    final CompletableFuture<Void> completableFuture = new CompletableFuture<>();
+    PubSubProducerCallback callback = new CompletableFutureCallback(completableFuture);
+
+    return sendMessage(
+        producerMetadata -> divKey,
+        MessageType.GLOBAL_RT_DIV,
+        putPayload,
+        partition,
+        callback,
+        DEFAULT_LEADER_METADATA_WRAPPER,
+        APP_DEFAULT_LOGICAL_TS);
   }
 
-  private CompletableFuture<PubSubProduceResult> sendDivMessageChunked(
+  private CompletableFuture<PubSubProduceResult> sendChunkedGlobalRtDivMessage(
       int partition,
       byte[] serializedKey,
-      byte[] serializedValue,
-      int valueSchemaId,
-      PutMetadata putMetadata) {
-    int replicationMetadataPayloadSize = putMetadata == null ? 0 : putMetadata.getSerializedSize();
-    final Supplier<String> reportSizeGenerator =
-        () -> getSizeReport(serializedKey.length, serializedValue.length, replicationMetadataPayloadSize);
-    // TODO: this needs to be changed later to adapt to div purpose.
+      byte[] serializedValue) {
+    final Supplier<String> reportSizeGenerator = () -> getSizeReport(serializedKey.length, serializedValue.length, 0);
+    // TODO: This needs to be implemented later to support Global RT DIV.
     final CompletableFuture<Void> completableFuture = new CompletableFuture<>();
     PubSubProducerCallback callback = new ErrorPropagationCallback(new CompletableFutureCallback(completableFuture));
-    BiConsumer<KeyProvider, Put> sendMessageFunction = (keyProvider, putPayload) -> sendMessage(
+    BiConsumer<KeyProvider, Object> sendMessageFunction = (keyProvider, putPayload) -> sendMessage(
         keyProvider,
-        MessageType.PUT,
+        MessageType.GLOBAL_RT_DIV,
```

Comment:
Is this correct? This is the KME type, and it should be `PUT` as you described in the description. Please also verify other places you set the KME type.

Reply:
Good catch. Fixed.